R/metadata.R

```diff
@@ -114,57 +115,91 @@ metadata <- function(surveyID,
   # RESHAPE DATA ----
 
   # Metadata
-  metadata <- data.frame(
-    "surveyID" = resp_filt$id,
-    "name" = resp_filt$name,
-    "ownerId" = resp_filt$ownerId,
-    "organizationId" = resp_filt$organizationId,
-    "isActive" = resp_filt$isActive,
-    "creationDate" = resp_filt$creationDate,
-    "lastModifiedDate" = resp_filt$lastModifiedDate,
-    "expiration_startdate" = ifelse(is.null(resp_filt$expiration$startDate),
-      NA,
-      resp_filt$expiration$startDate
-    ),
-    "expiration_endDate" = ifelse(is.null(resp_filt$expiration$endDate),
-      NA,
-      resp_filt$expiration$endDate
+  if (survey_definition) {
+    metadata <- data.frame(
+      "surveyID" = resp_filt$SurveyID,
+      "name" = resp_filt$SurveyName,
+      "ownerId" = resp_filt$OwnerID,
+      "organizationId" = resp_filt$CreatorID,
+      "isActive" = resp_filt$SurveyStatus
+    )
+
+    if (!is.null(q_select)) {
+      qnames <- vapply(resp_filt$Questions, function(x) {
+        x$DataExportTag
+      }, "")
+      if (all(q_select %in% qnames)) {
+        questions <- resp_filt$Questions[which(qnames %in% q_select)]
+      } else {
+        warning(paste0("One or more questions you queried are not present in your survey.\nReturning all questions instead.")) # nolint
+        questions <- resp_filt$Questions
+      }
+    } else {
+      questions <- resp_filt$Questions
+    }
+
+    # Construct metadata
+    met <- list(
+      "metadata" = metadata,
+      "questions" = questions,
+      "blocks" = resp_filt$Blocks,
+      "flow" = resp_filt$SurveyFlow
```

Comment:
This contains some variables (like `flow` and `blocks`) that are labeled `FALSE` in the `standard_list` in this function and don't get kept in what gets returned from this function. Is this intentional? It seems like you would want these, right? We'll need to carefully think through which objects to return.